### PR TITLE
Convert dependency management related subprojects to kotlin-dsl

### PR DESCRIPTION
**Pull request by jjohannes**
This PR also fixes the dependency declarations wrt. api/impl separation for these and related projects.
It fixes two places in production/test code where dependencies leaked into the classpath and the wrong classes where imported.

### CODE REVIEW

**Comment by eskatos** [comment]
@jjohannes given #4486 is currently on hold, could you rework this PR to not depend on it so this one can move forward?

btw, I updated https://github.com/gradle/gradle/issues/4649 to mention this PR

**Comment by jjohannes** [comment]
@eskatos I did what you suggested. This is now based on master. Would you mind doing a quick review?

**Comment by jjohannes** [comment]
Thanks @eskatos! I must have lost that when I rebased this PR. I removed the (three) `.gradle` files.

File: subprojects/dependency-management/dependency-management.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+testFixtures {
+    from(":core")
+    from(":messaging")
+    from(":modelCore")
+    from(":versionControl")
+}
+
+val verifyTestFilesCleanup by tasks.getting(EmptyDirectoryCheck::class) {
```
Comment (eskatos) [inline subprojects/dependency-management/dependency-management.gradle.kts]: :x: should use the `testFilesCleanup` extension instead:
```kotlin
testFilesCleanup {
    isErrorWhenNotEmpty = false
}
```

File: subprojects/ivy/ivy.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    moduleType = ModuleType.PLUGIN
+}
+
+testFixtures {
+    from(":core")
+    from(":modelCore")
+    from(":platformBase")
+}
+
+val verifyTestFilesCleanup by tasks.getting(EmptyDirectoryCheck::class) {
```
Comment (eskatos) [inline subprojects/ivy/ivy.gradle.kts]: :x: should use the `testFilesCleanup` extension instead:
```kotlin
testFilesCleanup {
    isErrorWhenNotEmpty = false
}
```

File: subprojects/dependency-management/dependency-management.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import org.gradle.build.ClasspathManifest
+import org.gradle.gradlebuild.testing.integrationtests.cleanup.EmptyDirectoryCheck
+import org.gradle.gradlebuild.unittestandcompile.ModuleType
+
+plugins {
+    `java-library`
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_6
```
Comment (eskatos) [inline subprojects/dependency-management/dependency-management.gradle.kts]: ❓This wasn't set in the Groovy script, why the change?

File: subprojects/maven/maven.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+gradlebuildJava {
+    moduleType = ModuleType.PLUGIN
+}
+
+testFixtures {
+    from(":core")
+    from(":modelCore")
+}
+
+val verifyTestFilesCleanup by tasks.getting(EmptyDirectoryCheck::class) {
```
Comment (eskatos) [inline subprojects/maven/maven.gradle.kts]: :x: should use the `testFilesCleanup` extension instead:
```kotlin
testFilesCleanup {
    isErrorWhenNotEmpty = false
}
```

File: subprojects/dependency-management/dependency-management.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import org.gradle.build.ClasspathManifest
+import org.gradle.gradlebuild.testing.integrationtests.cleanup.EmptyDirectoryCheck
+import org.gradle.gradlebuild.unittestandcompile.ModuleType
+
+plugins {
+    `java-library`
```
Comment (eskatos) [inline subprojects/dependency-management/dependency-management.gradle.kts]: \o/

File: subprojects/ivy/ivy.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import org.gradle.gradlebuild.testing.integrationtests.cleanup.EmptyDirectoryCheck
+import org.gradle.gradlebuild.unittestandcompile.ModuleType
+
+plugins {
+    `java-library`
```
Comment (eskatos) [inline subprojects/ivy/ivy.gradle.kts]: \o/

File: subprojects/maven/maven.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import org.gradle.gradlebuild.testing.integrationtests.cleanup.EmptyDirectoryCheck
+import org.gradle.gradlebuild.unittestandcompile.ModuleType
+
+plugins {
+    `java-library`
```
Comment (eskatos) [inline subprojects/maven/maven.gradle.kts]: \o/

File: subprojects/dependency-management/dependency-management.gradle.kts (hunk 1 above)
Comment (jjohannes) [reply subprojects/dependency-management/dependency-management.gradle.kts]: Wasn't aware of that. Thanks. Changed it.

File: subprojects/ivy/ivy.gradle.kts (hunk 2 above)
Comment (jjohannes) [reply subprojects/ivy/ivy.gradle.kts]: Changed.

File: subprojects/dependency-management/dependency-management.gradle.kts (hunk 3 above)
Comment (jjohannes) [reply subprojects/dependency-management/dependency-management.gradle.kts]: Damn. Another merging issue. This is now covered by `moduleType = ModuleType.WORKER`. Thanks for the thorough check.

File: subprojects/maven/maven.gradle.kts (hunk 4 above)
Comment (jjohannes) [reply subprojects/maven/maven.gradle.kts]: Changed.

**Comment by eskatos** [comment]
Feedback build https://builds.gradle.org/viewQueued.html?itemId=11729657
